app/src/common/shared/org/mozilla/vrbrowser/VRBrowserActivity.java

```diff
@@ -145,6 +147,7 @@ public void run() {
     LinkedList<PermissionListener> mPermissionListeners;
     LinkedList<FocusChangeListener> mFocusChangeListeners;
     LinkedList<WorldClickListener> mWorldClickListeners;
+    CopyOnWriteArrayList<Delegate> mConnectivityListeners;
```

Comment:
Why use a `CopyOnWriteArrayList`?

Reply:
Just to avoid concurrent modification issues. We actually talked about auditing all the app listeners and migrate the risky one, this might be low risk though.